### ISSUE TITLE
Clippy on macOS: dyld: Library not loaded: @rpath/librustc_driver-5d5e1e2505841b99.dylib

### ISSUE THREAD

**Issue by ajeetdsouza** [oxalica/rust-overlay#64]
Clippy throws this error on macOS CI:

```
>>> "cargo" "clippy" "--all-features" "--all-targets" "--" "-Dwarnings"
error: failed to run `rustc` to learn about target-specific information

Caused by:
  process didn't exit successfully: `/nix/store/r7lzrrkq759ak62a2sqyhmcsffn9xamp-rust-default-1.57.0/bin/clippy-driver rustc - --crate-name ___ --print=file-names --crate-type bin --crate-type rlib --crate-type dylib --crate-type cdylib --crate-type staticlib --crate-type proc-macro --print=sysroot --print=cfg` (signal: 6, SIGABRT: process abort signal)
  --- stderr
  dyld: Library not loaded: @rpath/librustc_driver-5d5e1e2505841b99.dylib
    Referenced from: /nix/store/r7lzrrkq759ak62a2sqyhmcsffn9xamp-rust-default-1.57.0/bin/clippy-driver
    Reason: image not found
```

rust-overlay commit ID: ad311f5bb5c5ef475985f1e0f264e831470a8510

[Link to the failing CI job](https://github.com/ajeetdsouza/zoxide/runs/4419738249)

**Comment by veehaitch** [comment]
Same here (also on macOS). According to `git bisect`, f5a5eddda1a64dca6830124aff5e09674e585477 causes the issue.

**Comment by oxalica** [comment]
Fixed in bb5adbd2a830e00aa783d9189f1f4183182663fd

**Comment by veehaitch** [comment]
Wow, thanks for the quick fix!
